pkg/reconciler/autoscaling/hpa/resources/hpa.go

```diff
@@ -82,6 +82,17 @@ func MakeHPA(pa *autoscalingv1alpha1.PodAutoscaler, config *autoscalerconfig.Con
 				},
 			}}
 		}
+	default:
+		if target, ok := pa.Target(); ok {
+			targetQuantity := resource.NewQuantity(int64(target), resource.DecimalSI)
+			hpa.Spec.Metrics = []autoscalingv2beta1.MetricSpec{{
+				Type: autoscalingv2beta1.PodsMetricSourceType,
+				Pods: &autoscalingv2beta1.PodsMetricSource{
+					MetricName:         pa.Metric(),
+					TargetAverageValue: *targetQuantity,
```

Comment:
is the assumption that we only want this to be `TargetAverageValue`? would we ever want `TargetValue` or `TargetAverageUtilization `?

Reply:
Sort of, yes. This seems like the most simple option out of the three. I wanted a simple change without adding new annotations. If we add an annotation we can allow the user to choose between them.

Reply:
I'm not sure how it would clear for users that its only using the TargetAverage, can we at least document it somehow

Reply:
Yes, that is right. I will add it here : https://github.com/knative/docs/blob/main/docs/serving/autoscaling/autoscaling-metrics.md

Reply:
@nader-ziada https://github.com/knative/docs/pull/4496